webhooks-extension/src/components/WebhookDisplayTable/WebhookDisplayTable.js

```diff
@@ -80,6 +135,94 @@ export class WebhookDisplayTable extends Component {
     }
   }
 
+  componentDidMount() {
+    this.getWebhooks();
+  }
+
+  async getWebhooks() {
+    const webhookData = await getWebhooks();
+    this.setState({
+      isLoaded: true,
+      webhooks: webhookData
+    });
+  }
+
+	showDeleteDialogHandlerInvisible = () => {
+		this.setState({
+			showDeleteDialog: false
+		});
+  }
+
+  showDeleteDialogHandlerVisible = rowsInput => {
+		this.setState({
+			showDeleteDialog: true,
+      userSelectedRows: rowsInput,
+    });
+	}
+
+  handleDeleteWebhook = () => {
+    let deleteRuns = false
+
+    if (this.state.checked) {
+        deleteRuns = true
+    };
+
+    let rowsToUse = getSelectedRows(this.state.userSelectedRows)
+
+    let deletePromises = []
+
+    deletePromises = rowsToUse.map(function(rowIDObject, indexInTheMap) {
+      let id = rowIDObject.id
+      let theName = id.substring(0, id.lastIndexOf('|'))
+      let namespace = id.substring(id.lastIndexOf('|') + 1, id.length)
+      let response = deleteWebhooks(theName, namespace, deleteRuns)
+      let theTimeoutPeriod = 500
+      let theTimeout = new Promise((resolve, reject) => {
+        setTimeout(function () {
+          reject()
+        }, theTimeoutPeriod)
+      })
+      let myRace = Promise.race([response, theTimeout])
+      return myRace
+    })
+
+    Promise.all(deletePromises).then(values => {
+      this.getWebhooks();
+      this.setState({
+        deleteEvent: true,
+        showDeleteDialog: false,
+        deleteStatus: 'success',
+        deleteErrorMessage: '',
+        deleteStatusPretty: 'Webhook(s) deleted successfully.',
+      })
+     }).catch(error => {
+      this.setState({
+        deleteEvent: true,
+        showDeleteDialog: false,
+        deleteStatus: 'error',
+        // Todo can we get the actual API error instead?
```

Comment:
Yes ..... look at the create webhook js ..... the actual API errors are used there.